flink-runtime/src/main/java/org/apache/flink/runtime/net/NetUtils.java

```diff
@@ -189,9 +191,17 @@ public static InetAddress findConnectingAddress(InetSocketAddress targetAddress,
 		long currentSleepTime = MIN_SLEEP_TIME;
 		long elapsedTime = 0;
 
+		// before trying with different strategies: test with getLocalHost():
+		InetAddress localhostName = InetAddress.getLocalHost();
+
+		if(tryToConnect(localhostName, targetAddress, AddressDetectionState.ADDRESS.getTimeout(), false)) {
+			LOG.debug("Using immediately InetAddress.getLocalHost() for the connecting address");
```

Comment:
also it could throw `UnknownHostException` - not sure if that could be a problem